developer_guides/firmware/kd_integration/kd-integration.rst

```diff
@@ -114,4 +115,44 @@ keyphrase detection flows. The components are organized in pipelines:
       information if a keyphrase is detected. Note that the FW infrastructure
       can allow a FW event to be sent to the Keyphrase Buffer Manager
       component if keyphrase is detected. The component also sends a 
-      notification to the audio driver and implements large parameters support.
\ No newline at end of file
+      notification to the audio driver and implements large parameters support.
+
+Latency & buffering
+*******************
+
+This section covers calculations needed to be done to properly configure
+keyphrase buffer size. The symbols used in a formula below are depicted
+above, see :ref:`timing-sequence`.
+
+.. note::
+
+   The formula for size of a keyphrase buffer:
+   ( L1 + L2 + L3 + L4 ) * number of channels * bitdepth = Size [Kb]
+
+
+Specifically:
+
+1. L1 is defined as length of a keyphrase with preceding or trailing silence,
+   value depends highly on the keyphrase itself and detection algorithm
+   requirements.
+
+2. L2 is a sum of algorithmic (processing) latency of a detection algorithm and
+   additional time needed to execute additional components in pipelines,
+   prepare and send notifications.
+
+3. L3 is a time required to send already buffered data to host. Typically
+   Write Pointer (WP) is used to indicate where data coming from microphones is
+   written to a keyphrase buffer. The keyphrase buffer is organized as
+   a cyclic buffer and WP moves as long as data is coming from mics at
+   a regular rate. Read pointer (RP) indicates from which offset in teh buffer
+   data is fetched to host. To start burst transmission RP is set to WP
+   position. RP moves faster then WP as draining is executed as a background
+   task until RP reaches again WP that moves a a regular (slower) rate. This
```

Comment:
typo, s/a/at

Reply:
Also, here we claim that we start draining from WP (write pointer) which is not true - in fact we calculate read position base on host param called "history depth".

Reply:
@lbetlej Please let me know when/if you make this update based on @mmaka1 's remarks. Thanks

Reply:
@lbetlej @lgirdwood Please let me know if this comment by @mrajwa has been resolved so I can merge this PR.  [...in fact we calculate read position base on host client algorithm / topology param called "history depth"]

Thanks.

Reply:
@mrajwa please respond.

Reply:
@lgirdwood there is nothing more I can say/do in regards to this PR. We have to wait for an update from @lbetlej. He is on sick leave now so let's give him few days until his well.

Reply:
@mrajwa Thanks for your update.  In the interest of time, I am going to merge PRs 182-183 and also add my own edits.  I will make a note of @mrajwa's comments. @lbetlej can update the new page when he returns. Thanks, all.

Reply:
@deb-intel I've rebased and updated the PR to address all findings listed in the conversation. Could you wait for review approval from @mrajwa before you start proof reading ?